### PR TITLE
8280235: Deprecated flag FlightRecorder missing from VMDeprecatedOptions test

### PR DESCRIPTION
**Pull request by tkiriyama**
I would like to fix 8280235: Deprecated flag FlightRecorder missing from VMDeprecatedOptions test. 

FlightRecorder option has not been tested since JDK13.
I think we should test it, because FlightRecorder option has not been obsolete in the latest JDK.
Users would be in trouble if the option suddenly disappears without notice, 
so it's important to confirm the deprication message.

Also we should add a test of ExtendedDTraceProbes option.
The test was disabled in 8281675, because some jdk can't specify it.
I modified the test to be able to verify  ExtendedDTraceProbes in either case that DTRACE_ENABLED is enabled or not.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8280235](https://bugs.openjdk.org/browse/JDK-8280235): Deprecated flag FlightRecorder missing from VMDeprecatedOptions test

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Markus Grönlund](https://openjdk.org/census#mgronlun) (@mgronlun - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9123/head:pull/9123` \
`$ git checkout pull/9123`

Update a local copy of the PR: \
`$ git checkout pull/9123` \
`$ git pull https://git.openjdk.org/jdk pull/9123/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9123`

View PR using the GUI difftool: \
`$ git pr show -t 9123`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9123.diff">https://git.openjdk.org/jdk/pull/9123.diff</a>

</details>

### CODE REVIEW

**Comment by dholmes-ora** [comment]
> so it's important to confirm the deprication message.

It has been confirmed, just not by a regression test. Adding this is harmless _but_ you can only test this on a JVM configured with JFR so now the test would have to `@require vm.jfr` as well.

> Also we should add a test of ExtendedDTraceProbes option.

That might be useful in 19 but this bug is being pushed to 20 and it incorrect for 20. At the start of every release we remove  the options present in the VMDeprecatedOptions test that are obsoleted in that release as the error message changes and so would cause the test to fail.

**Comment by tkiriyama** [comment]
@dholmes-ora 
Thank you for your review.
I modified the test to check whether JFR is present or not, 
and deleted the test of ExtendedDTraceProbes option.
Could you please review the fix.

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java:103]: This is not the way to handle this. In the static block that initializes the set of flags to test you should do something like:

```
if (wb.isJFRIncluded()) {
  deprecated.add("FlightRecorder", false);
}
```
before line 70.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java
Comment (tkiriyama) [reply test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java]: Thank you for your comment. 
I fixed the test. Could you please review it again?

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java:71]: No need for all the extra whitespace after the comma - just a single space please.

**Review by dholmes-ora — APPROVED** [review]
Thanks for the update. Small style nit but otherwise good.

Thanks.

File: test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java
Comment (tkiriyama) [reply test/hotspot/jtreg/runtime/CommandLine/VMDeprecatedOptions.java]: That's right. I fixed it.

**Review by dholmes-ora — APPROVED** [review]

**Comment by tkiriyama** [comment]
@dholmes-ora 
Could you sponsor this fix, please?

**Comment by dholmes-ora** [comment]
This needs a second review as per hotspot rules.

**Review by mgronlun — APPROVED** [review]

**Pull request closed without merging**
